frontend/src/scenes/organization/membersLogic.tsx

```diff
@@ -63,40 +143,46 @@ export const membersLogic = kea<membersLogicType>([
             },
         ],
         membersFuse: [
-            (s) => [s.members],
+            (s) => [s.meFirstMembers],
             (members): MembersFuse =>
-                new Fuse<OrganizationMemberType>(members, {
+                new Fuse<OrganizationMemberType>(members ?? [], {
                     keys: ['user.first_name', 'user.last_name', 'user.email'],
                     threshold: 0.3,
                 }),
         ],
         filteredMembers: [
-            (s) => [s.members, s.membersFuse, s.search],
-            (members, membersFuse, search) =>
-                search ? membersFuse.search(search).map((result) => result.item) : members,
+            (s) => [s.meFirstMembers, s.membersFuse, s.search],
+            (members, membersFuse, search): OrganizationMemberType[] =>
+                search ? membersFuse.search(search).map((result) => result.item) : members ?? [],
+        ],
+        memberCount: [
+            (s) => [s.user, s.sortedMembers],
+            (user, members): number => {
+                // Typically we can rely on the app context but just in case we use the loaded members if present
+                const count = user?.organization?.member_count
+                return Math.max(count ?? 0, members?.length ?? 0)
+            },
         ],
     }),
-    listeners(({ actions }) => ({
-        changeMemberAccessLevel: async ({ member, level }) => {
-            await api.update(`api/organizations/@current/members/${member.user.uuid}/`, { level })
-            lemonToast.success(
-                <>
-                    Made <b>{member.user.first_name}</b> organization {membershipLevelToName.get(level)}
-                </>
-            )
-            // reload organization to account for no longer being organization owner
-            if (level === OrganizationMembershipLevel.Owner) {
-                organizationLogic.actions.loadCurrentOrganization()
-            }
-            actions.loadMembers()
-        },
+
+    listeners(({ values, actions }) => ({
         postRemoveMember: async ({ userUuid }) => {
             if (userUuid === userLogic.values.user?.uuid) {
                 location.reload()
             }
         },
+
+        ensureAllMembersLoaded: async () => {
+            if (values.membersLoading) {
+                return
+            }
+            if (!values.members) {
+                actions.loadAllMembers()
+            } else {
+                actions.loadMemberUpdates()
+            }
+        },
```

Comment:
For some reason this function seems unnecessary to me. Why don't we just call `loadMembers()` and do these checks inside that function? I can see a somewhat weird situation where from that listener you either get all members or return the current list and then do the call for updates. But i think it would be less confusing to always just call loadMembers() when you need to make sure they are up to date.

Reply:
It's a pattern we use a lot to separate the logic of deciding if something should be loaded from the actual loading.

If we would do it within `loadMembers` then all the associated things would fire `membersLoading` `loadMembersSuccess` etc. which doesn't make sense (at least in my mind). This keeps it clean